sympy/polys/monomials.py

```diff
@@ -274,6 +274,35 @@ def monomial_gcd(A, B):
     """
     return tuple([ min(a, b) for a, b in zip(A, B) ])
 
+def monomial_ngcd(monomials):
+    """
+    Computes the greatest common divisor (GCD) of the exponents for each
+    variable in the monomials.
+
+    Parameters
+    ==========
+
+    monomials : iterable of tuples
+        Represents monomials, where each monomial is a tuple of exponents for variables.
+
+    Returns
+    =======
+
+    tuple
+        The greatest common divisor of the exponents for each variable in the
+        monomials.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.monomials import monomial_ngcd
+    >>> monomial_ngcd({(1, 0), (0, 2), (2, 0), (0, 1)})
+    (0, 0)
+
+    """
+    monomial_ngcd = tuple(map(min, zip(*monomials)))
+    return monomial_ngcd
```

Comment:
The optimisations discussed previously for the zero monom case could be applied here. I expect that a very common case will be a large number of monomials whose gcd is just 1 and such that the gcd of the first few monomials would be revealed to be 1. We could probably detect that without needing to loop very far through the long list of monomials.

Another common case would be a large number of monomials that *does* have a nontrivial gcd. Using `map(min, zip(*monomials))` micro-optimises for this second case at the expense of a more important optimisation for the first case but I think that the first case is likely to be very common.

In the first instance I would write this as a plain loop like:
```python
# This code needs to be adapted for the inputs the function currently takes
for monom in monomials:
    gcd = monomial_gcd(gcd, monom)
    if gcd == zero_monom:
        # trivial gcd: exit early
        return gcd

# A nontrivial gcd was found:
return gcd
```
Timings should be done to see if the `map,min,zip` approach offers any improvement over this but the loop with early exit would be much faster in the common case of many (effectively) random monomials.